test/passes/issue2788.wast

```diff
@@ -0,0 +1,171 @@
+;; Regression test for 'std::move'-related issues in ExpressionRunner during precompute-propagate
```

Comment:
Can we possibly reduce this test case? Do we need all this code to reproduce this bug?

Reply:
Attempts to far didn't work so well (see also https://github.com/WebAssembly/binaryen/pull/2790#pullrequestreview-398686088), but I can try a little more if you'd like. My current guess is that this depends on the implementation of `std::unordered_set` a bit, i.e. how and when it shuffles its buffers around, and reducing it scares off the bug.

Reply:
Interesting. I think in that case I'd prefer to not check in the testcase, if it's not going to deterministically hit the bug anyhow.

I think we're ok without a testcase since the fuzzer found it easily multiple times for me.